sds/src/proximity_keywords/mod.rs

```diff
@@ -279,19 +279,29 @@ fn get_char_type(c: &char) -> CharType {
 
 /// Function that standardizes a list of characters, by pushing characters one by one in a standard way.
 /// Takes a closure that will be called when a character is to be pushed
-pub fn standardize_path_chars<F>(chars: &str, mut push_character: F)
+pub fn standardize_path_chars<F>(characters: &str, mut push_character: F)
 where
     F: FnMut(&char),
 {
-    let kw_length = chars.len();
+    let kw_length = characters.len();
+    if kw_length == 0 {
+        return;
+    }
 
-    if let Some(c) = chars.chars().next() {
-        push_character(&c);
-    } else {
+    let mut char_iterator = characters.chars();
+    if char_iterator.all(|c| c.is_ascii_lowercase())
+        || char_iterator.all(|c| c.is_ascii_uppercase())
+    {
+        for c in characters.chars() {
+            push_character(&c)
+        }
         return;
     }
 
-    for (i, chars) in chars.as_bytes().windows(2).enumerate() {
+    let character_bytes = characters.as_bytes();
```

Comment:
Why is `characters` being converted to bytes, and then back to `chars`? (It was already like this before but I didn't notice it). This seems both unnecessary and potentially dangerous since it's assuming all characters are a single byte.

Reply:
Yep! I agree it's dangerous: [Remove .as_bytes and use the iterator instead](https://github.com/DataDog/dd-sensitive-data-scanner/pull/74/commits/329ec79641dd5e707c6ef3347071fd19f2d065fc)